TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/Models/PushFactor.swift

```diff
@@ -20,17 +20,95 @@
 import Foundation
 
 struct PushFactor: Factor, Codable {
+
   var status: FactorStatus = .unverified
   let sid: String
   let friendlyName: String
   let accountSid: String
   let serviceSid: String
   let identity: String
-  let type: FactorType = .push
+  var type: FactorType = .push
+  var allowIphoneMigration: Bool = false
   let createdAt: Date
   let config: Config
   var keyPairAlias: String?
   var metadata: [String: String]?
+
+  init(
+    status: FactorStatus = .unverified,
+    sid: String,
+    friendlyName: String,
+    accountSid: String,
+    serviceSid: String,
+    identity: String,
+    type: FactorType = .push,
+    allowIphoneMigration: Bool = false,
+    createdAt: Date,
+    config: Config,
+    keyPairAlias: String? = nil,
+    metadata: [String : String]? = nil
+  ) {
+    self.status = status
+    self.sid = sid
+    self.friendlyName = friendlyName
+    self.accountSid = accountSid
+    self.serviceSid = serviceSid
+    self.identity = identity
+    self.type = type
+    self.allowIphoneMigration = allowIphoneMigration
+    self.createdAt = createdAt
+    self.config = config
+    self.keyPairAlias = keyPairAlias
+    self.metadata = metadata
+  }
+
+  enum CodingKeys: CodingKey {
+    case status
+    case sid
+    case friendlyName
+    case accountSid
+    case serviceSid
+    case identity
+    case type
+    case allowIphoneMigration
+    case createdAt
+    case config
+    case keyPairAlias
+    case metadata
+  }
+
+  init(from decoder: Decoder) throws {
+    let container: KeyedDecodingContainer<PushFactor.CodingKeys> = try decoder.container(keyedBy: PushFactor.CodingKeys.self)
+    self.status = try container.decode(FactorStatus.self, forKey: PushFactor.CodingKeys.status)
+    self.sid = try container.decode(String.self, forKey: PushFactor.CodingKeys.sid)
+    self.friendlyName = try container.decode(String.self, forKey: PushFactor.CodingKeys.friendlyName)
+    self.accountSid = try container.decode(String.self, forKey: PushFactor.CodingKeys.accountSid)
+    self.serviceSid = try container.decode(String.self, forKey: PushFactor.CodingKeys.serviceSid)
+    self.identity = try container.decode(String.self, forKey: PushFactor.CodingKeys.identity)
+    self.type = try container.decodeIfPresent(FactorType.self, forKey: PushFactor.CodingKeys.type) ?? .push
+    self.allowIphoneMigration = try container.decodeIfPresent(Bool.self, forKey: PushFactor.CodingKeys.allowIphoneMigration) ?? false
+    self.createdAt = try container.decode(Date.self, forKey: PushFactor.CodingKeys.createdAt)
+    self.config = try container.decode(Config.self, forKey: PushFactor.CodingKeys.config)
+    self.keyPairAlias = try container.decodeIfPresent(String.self, forKey: PushFactor.CodingKeys.keyPairAlias)
+    self.metadata = try container.decodeIfPresent([String : String].self, forKey: PushFactor.CodingKeys.metadata)
```

Comment:
<table data-meta="generated_by_UnitTests">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/twilio/twilio-verify-ios/blob/eb579290ef458a5ac85d4e99b29e765308b08746/TwilioVerifySDK/TwilioVerify/Sources/Domain/Factor/Models/PushFactor.swift#L93"></span>Colons should be next to the identifier when specifying a type and next to the key in dictionary literals.<br />
<code>colon</code> <code>PushFactor.swift:93</code></td>
    </tr>
  </tbody>
</table>